### DISCUSSION TITLE
Why doesn't nz-input-number use type number

### DISCUSSION THREAD

**Discussion by saif317** [NG-ZORRO/ng-zorro-antd#7655] [Q&A] [answered]
in the input-number.component.ts file, this is how the input element is defined:

```
<div class="ant-input-number-input-wrap">
      <input
        #inputElement
        autocomplete="off"
        class="ant-input-number-input"
        [attr.id]="nzId"
        [attr.autofocus]="nzAutoFocus ? 'autofocus' : null"
        [disabled]="nzDisabled"
        [attr.min]="nzMin"
        [attr.max]="nzMax"
        [placeholder]="nzPlaceHolder"
        [attr.step]="nzStep"
        [readOnly]="nzReadOnly"
        [attr.inputmode]="nzInputMode"
        [ngModel]="displayValue"
        (ngModelChange)="onModelChange($event)"
      />
</div>
```
the type of the input element is not defined so it's defaulting to text, this allows the user to input non-numeric characters into the field, ng zorro handles that by removing non-numeric characters from the field, but why not use the default behavior of not allowing the user to input non-numeric character at all.

any specific reason why the component was designed this way?

**Comment by kkarkal** [comment]
I'm just checking this issue and am wondering the same. Basically when the doc says `nz-input-numeric`, It by default should take only numeric characters provided we already have `nz-input` for dealing with non-numeric characters. I want to know the reasoning behind this.

**Comment by ParsaArvanehPA** [comment]
Hey guys, @kkarkal @saif317 

I'm not entirely certain why the initial decision was made against using type="number". However, I can offer some educated guesses:
1. The behavior of the input type="number" varies significantly different across browsers. For instance, Chrome displays a numeric keypad on mobile devices and restricts input to digits only, but this is not consistent in all browsers. For a more detailed explanation, you might find this [video by Kevin Powell](https://www.youtube.com/watch?v=nnZS761ngXE) informative.

2. If the input type is set to number, it precludes the possibility of modifying the user's input. Take, for example, the implementation of a three-digit separator; while the `nzFormatter` could be utilized for this purpose, an input with type number would not accept commas.

I hope this provides some clarity on the subject.

**Comment by HyperLife1119** [comment, marked as answer]
Since this component needs to support formatter/parser, the input box should be of text type, not just numbers :)
